### PR TITLE
feat: `avm/res/data-protection/backup-vault`

### PR DESCRIPTION
**Pull request by segraef**
[avm.res.data-protection.backup-vault](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.data-protection.backup-vault.yml)

### CODE REVIEW

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Requires uncommenting in the `/.github/CODEOWNERS` file

**Comment by segraef** [comment]
> Requires uncommenting in the `/.github/CODEOWNERS` file

done

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #/avm/res/databricks/access-connector/ @Azure/avm-res-databricks-accessconnector-module-owners-bicep @Azure/avm-core-team-technical-bicep
 #/avm/res/databricks/workspace/ @Azure/avm-res-databricks-workspace-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/data-factory/factory/ @Azure/avm-res-datafactory-factory-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/data-protection/backup-vault/ @Azure/avm-res-dataprotection-backupvault-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/data-protection/backup-vault/ @Azure/avm-res-dataprotection-backupvault-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (AlexanderSehr) [inline .github/CODEOWNERS:40]: Team seems to be missing

Comment (eriqua) [reply .github/CODEOWNERS:40]: And added to parent team as per https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships

File: .github/workflows/avm.res.data-protection.backup-vault.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  ###########################
+  #   Initialize pipeline   #
+  ###########################
+  job_initialize_pipeline:
+    runs-on: ubuntu-20.04
```
Comment (eriqua) [inline .github/workflows/avm.res.data-protection.backup-vault.yml:47]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/CODEOWNERS (hunk 1 above)
Comment (segraef) [reply .github/CODEOWNERS:40]: Exist already:
@Azure/avm-res-consumption-budget-module-owners-bicep
@Azure/avm-res-consumption-budget-module-contributors-bicep

File: avm/res/data-protection/backup-vault/main.bicep [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+output resourceId string = backupVault.id
+
+@description('The name of the resource group the recovery services vault was created in.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The Name of the backup vault.')
+output name string = backupVault.name
+
+@description('The principal ID of the system assigned identity.')
+output systemAssignedMIPrincipalId string = (managedIdentities.?systemAssigned ?? false) && contains(backupVault.identity, 'principalId') ? backupVault.identity.principalId : ''
```
Comment (AlexanderSehr) [inline avm/res/data-protection/backup-vault/main.bicep:155]:
```suggestion
output systemAssignedMIPrincipalId string = backupVault.?identity.?principalId ?? ''
```

File: .github/CODEOWNERS (hunk 1 above)
Comment (AlexanderSehr) [reply .github/CODEOWNERS:40]: May still be missing the owner assignment / its approval

Comment (segraef) [reply .github/CODEOWNERS:40]: both teams exist and parent teams are assigned

Comment (AlexanderSehr) [reply .github/CODEOWNERS:40]: True true

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**